tasks/UWPoseEstimatorFixedCovariance.cpp

```diff
@@ -0,0 +1,121 @@
+/* Generated from orogen/lib/orogen/templates/tasks/Task.cpp */
+
+#include "UWPoseEstimatorFixedCovariance.hpp"
+
+using namespace pose_estimation;
+
+UWPoseEstimatorFixedCovariance::UWPoseEstimatorFixedCovariance(std::string const& name)
+    : UWPoseEstimatorFixedCovarianceBase(name)
+{
+}
+
+UWPoseEstimatorFixedCovariance::~UWPoseEstimatorFixedCovariance()
+{
+}
+
+/// The following lines are template definitions for the various state machine
+// hooks defined by Orocos::RTT. See UWPoseEstimatorFixedCovariance.hpp for more detailed
+// documentation about them.
+
+bool UWPoseEstimatorFixedCovariance::configureHook()
+{
+    if (!UWPoseEstimatorFixedCovarianceBase::configureHook())
+        return false;
+
+    m_cov_depth = _cov_depth.get();
+    m_cov_orientation = _cov_orientation.get();
+    return true;
+}
+bool UWPoseEstimatorFixedCovariance::startHook()
+{
+    if (!UWPoseEstimatorFixedCovarianceBase::startHook())
+        return false;
+    return true;
+}
+
+void UWPoseEstimatorFixedCovariance::updateHook()
+{
+    UWPoseEstimatorFixedCovarianceBase::updateHook();
+}
+
+void UWPoseEstimatorFixedCovariance::depth_samplesTransformerCallback(
+    const base::Time& ts,
+    const ::base::samples::RigidBodyState& depth_samples_sample)
+{
+    // receive sensor to body transformation
+    Eigen::Affine3d sensorInBody;
+    if (!getSensorInBodyPose(_pressure_sensor2body, ts, sensorInBody))
+        return;
+
+    PoseUKF::State current_state;
+    if (!base::isNaN(depth_samples_sample.position.z()) && !base::isNaN(m_cov_depth) &&
+        pose_estimator->getCurrentState(current_state)) {
+        predictionStep(ts);
```

Comment:
Same suggestion ... override the input and pass it to the base class' implementation ?